include/z64scene.h

```diff
@@ -813,6 +813,15 @@ typedef enum {
 */
 #define ENTRANCE(scene, spawn) ((((ENTR_SCENE_##scene) & 0x7F) << 9) | (((spawn) & 0x1F) << 4))
 
+/*
+* Entrances used in cutscene destination. Includes scene layer that's immediately applied to `nextCutsceneIndex` and removed.
+* See `CutsceneScriptEntry.nextEntrance`
+* 0xFE00:  Index into sSceneEntranceTable (Scene)
+* 0x01F0:  Index into the scenes specific entrance table (Spawn)
+* 0x000F:  Index into the specific entrance table (Layer)
+*/
+#define CS_ENTRANCE(scene, spawn, layer) ((((ENTR_SCENE_##scene) & 0x7F) << 9) | (((spawn) & 0x1F) << 4) | ((layer) & 0xF))
```

Comment:
I don't think is necessarily because we don't want to specify layer, it is that the main use for `ENTRANCE` is to set `play->nextEntrance` which shouldn't contain the `layer` due to things like
https://github.com/Decompollaborate/mm/blob/play/src/code/z_play.c#L566
in `play.c`.
Cutscenes on the other hand need their entrances to specify the `layer` as it gets used, as such I feel like the current names are fine.

Reply:
Yeah, that's what I've been meaning to test with this. The more general `ENTRANCE` macro is used like:
```c
play->nextEntrance = ENTRANCE(CUTSCENE, 0);
gSaveContext.nextCutsceneIndex = 0xFFF8; // sceneLayer 9
play->transitionTrigger = TRANS_TRIGGER_START;
play->transitionType = TRANS_TYPE_03;
```
Where `nextCutsceneIndex` is used to set this, which leads to the `sceneLayer` being set.

So if you did specify the spawnLayer in nextEntrance i.e. 
```c
play->nextEntrance = ENTRANCE(CUTSCENE, 0, 9); // sceneLayer 9
gSaveContext.nextCutsceneIndex = 0;
play->transitionTrigger = TRANS_TRIGGER_START;
play->transitionType = TRANS_TYPE_03;
```
Would that work as an alternative?

For reference in z_demo, the way the `nextEntrance` is set is by forcing `nextCutsceneIndex = 0`:

https://github.com/zeldaret/mm/blob/052d4d5081ba6e0d525bf6cec9e6a9cf19906109/src/code/z_demo.c#L594-L595

Then reads the scene layer here:
https://github.com/zeldaret/mm/blob/052d4d5081ba6e0d525bf6cec9e6a9cf19906109/src/code/z_demo.c#L608-L610

Then removes the scene layer from the macro here:
https://github.com/zeldaret/mm/blob/052d4d5081ba6e0d525bf6cec9e6a9cf19906109/src/code/z_demo.c#L612

Reply:
Okay, so I did some testing:
```c
play->nextEntrance = ENTRANCE(CUTSCENE, 0);
gSaveContext.nextCutsceneIndex = 0xFFF8;
play->transitionTrigger = TRANS_TRIGGER_START;
play->transitionType = TRANS_TYPE_03;
```
This is what you'd expect for the "Fire Wall" Cutscene - layer 9, spawn 0

```c
play->nextEntrance = ENTRANCE(CUTSCENE, 0);
gSaveContext.nextCutsceneIndex = 0;
play->transitionTrigger = TRANS_TRIGGER_START;
play->transitionType = TRANS_TYPE_03;
```
This is what'd you'd expect for the opening cs (first text)

Now doing both:
```c
play->nextEntrance = CS_ENTRANCE(CUTSCENE, 0, 9);
gSaveContext.nextCutsceneIndex = 0;
play->transitionTrigger = TRANS_TRIGGER_START;
play->transitionType = TRANS_TYPE_03;
```
and
```c
play->nextEntrance = CS_ENTRANCE(CUTSCENE, 0, 9);
gSaveContext.nextCutsceneIndex = 0xFFEF;
play->transitionTrigger = TRANS_TRIGGER_START;
play->transitionType = TRANS_TYPE_03;
```
Would expect this to be the "Fire Wall" cutscene (layer 9), but what actually happens is the opening cutscene plays (layer 0).

So actually including the `layer` in the entrance macro doesn't work in general, it seems specific to cutscenes. So that's why I'm thinking `CS_ENTRANCE` (to indicate it's cutscene specific) over `ENTRANCE_FULL` (which would indicate it could be used anywhere)